### PR TITLE
updating dev from alejandro

### PR DESCRIPTION
**Pull request by AlejandroMunozManterola**

### CODE REVIEW

File: src/driver/driver.cpp [hunk 1]
```diff
@@ -78,7 +78,7 @@ FieldType getFieldType(const std::string& ft)
 	}
 }
 
-Sources buildGaussianInitialField(
+std::unique_ptr<InitialField> buildGaussianInitialField(
```
Comment (lmdiazangulo) [inline src/driver/driver.cpp:81]: why did you change this? Is it not better to have a container for that?

File: src/evolution/EvolutionMethods.cpp [hunk 2]
```diff
@@ -15,12 +15,14 @@ FluxBdrCoefficientsCentered bdrCentCoeff{
 	{BdrCond::PEC               ,	{2.0, 0.0}},
 	{BdrCond::PMC               ,	{0.0, 2.0}},
 	{BdrCond::SMA               ,	{0.0, 0.0}},
+	{BdrCond::SurfaceCond       ,   {1.0, 1.0}},
```
Comment (lmdiazangulo) [inline src/evolution/EvolutionMethods.cpp:18]: Surface conductivity would appear as an Surface Impedance Boundary Condition (SIBC). Are you planning to implement that?

File: src/math/PhysicalConstants.h [hunk 3]
```diff
@@ -1,7 +1,14 @@
 #pragma once
 
+#define _USE_MATH_DEFINES
+#include <math.h>
+
 namespace maxwell {
 namespace physicalConstants {
 	constexpr double speedOfLight = 1.0;
+	const double true_vacuum_permeability = 4.0 * M_PI * 1e-7;
```
Comment (lmdiazangulo) [inline src/math/PhysicalConstants.h:9]: I would call this vacuumPermeabilitySI. It's more a matter of units rather than being true or false

File: test/maxwell/components/MaterialTest.cpp [hunk 4]
```diff
@@ -10,10 +10,10 @@ class MaterialTest : public ::testing::Test {
 
 TEST_F(MaterialTest, impedanceAndConductance)
 {
-	Material mat1(1.0, 2.0);
-	Material mat2(100.0, 1.0);
-	Material mat3(10.0, 20.0);
-	Material mat4(9.0, 30.0);
+	Material mat1(1.0  , 2.0 , 0.0);
```
Comment (lmdiazangulo) [inline test/maxwell/components/MaterialTest.cpp:13]: Rather than having to change this everywhere to include conductivuty, you could simply assign a default value of zero in the ctor declaration/definition.

**Review by lmdiazangulo — CHANGES_REQUESTED** [review]
I have made some comments.

File: src/math/PhysicalConstants.h (hunk 3 above)
Comment (AlejandroMunozManterola) [reply src/math/PhysicalConstants.h:9]: Agreed.

File: src/components/Material.h [hunk 5]
```diff
@@ -6,15 +6,16 @@ namespace maxwell {
...
-	Material(double epsilon, double mu);
+	Material(double epsilon, double mu, double sigma_);
 
 	double getPermittivity() const { return epsilon_; }
 	double getPermeability() const { return mu_; }
+	double getConductivity() const { return sigma_; }
 	double getImpedance() const { return sqrt(mu_ / epsilon_); }
 	double getAdmitance() const { return sqrt(epsilon_ / mu_); }
-	double getSpeedOfLight() const { return 1 / sqrt(mu_ * epsilon_); }
+	double getSpeedOfWave() const { return 1 / sqrt(mu_ * epsilon_); }
```
[the remark this answers is not in the document]
Comment (AlejandroMunozManterola) [reply src/components/Material.h:16]: Correct, I forgot to change this.

I will think about how to implement this. If the material is conductive the speed of the wave now depends on the conductivity and the frequency of the wave. Considering our time domain and our mainly gaussian sources, it wouldn't be possible to give a singular answer.

Any suggestions?

[the remark this answers is not in the document]
Comment (lmdiazangulo) [reply src/components/Material.h:16]: I would simply throw if conductivity is not zero. There is no simple answer.

File: src/evolution/EvolutionMethods.cpp (hunk 2 above)
Comment (lmdiazangulo) [reply src/evolution/EvolutionMethods.cpp:18]: I do not think that it is a simple number. That kind of conditions probably imply interconnecting with a 1D domain and/or possibly additional degrees of freedom.

If interconnecting with a 1D domain, probably the best is to use the standard numerical fluxes that normal propagation between elements use.

File: src/math/PhysicalConstants.h (hunk 3 above)
Comment (lmdiazangulo) [reply src/math/PhysicalConstants.h]: Could you change it?

**Review by lmdiazangulo — APPROVED** [review]
